src/modals/CRUD.tsx

```diff
@@ -0,0 +1,85 @@
+import ReactModal, { Styles } from "react-modal"
+import { ModalParams } from "../redux/global/global.types"
+import React, { useState } from "react"
+import gatewayClient from "../services/gatewayClient"
+import { useDispatch } from "react-redux"
+import { handleGatewayConnectionStatus } from "../redux/global/global.actions"
+import store from "../redux"
+
+const style: Styles = {
+  overlay: {
+    backgroundColor: "rgba(38, 50, 56, 0.5)",
+  },
+  content: {
+    border: "none",
+    bottom: "auto",
+    left: "50%",
+    position: "fixed",
+    right: "auto",
+    top: "50%", // start from center
+    transform: "translate(-50%,-50%)", // adjust top "up" based on height
+    maxWidth: "30rem",
+    overflow: "hidden",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+}
+
+type Props = {
+  open: boolean
+  closeModal: () => void
+  modalParams: ModalParams
+}
+
+function CRUD({ open, closeModal, modalParams }: Props) {
```

Comment:
Can we improve the name of this component? CRUD sounds very generic

Reply:
How about querySearchModal?

Reply:
Sounds good